### PR TITLE
bee-packable crate refactoring

### PR DESCRIPTION
**Pull request by thibault-martinez**
Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-common/bee-packable/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
-name = "bee-common"
-version = "1.0.0-alpha"
+name = "bee-packable"
+version = "0.1.0"
 authors = ["IOTA Stiftung"]
 edition = "2018"
 description = "Common utilities used across the bee framework"
```
Comment (pvdrz) [inline bee-common/bee-packable/Cargo.toml]: the description should change.

File: bee-common/bee-packable/src/packable.rs [hunk 2]
```diff
@@ -5,21 +5,17 @@
...
-mod io;
-mod packer;
-mod unpacker;
+pub use crate::{
+    error::{UnknownTagError, UnpackError},
+    packer::{Packer, VecPacker},
+    unpacker::{SliceUnpacker, UnexpectedEOF, Unpacker},
+};
+
+pub use bee_common_derive::Packable;
```
Comment (pvdrz) [inline bee-common/bee-packable/src/packable.rs]: should we have `bee-packable-derive` too?

**Review by pvdrz — CHANGES_REQUESTED** [review]

File: bee-common/bee-packable/Cargo.toml (hunk 1 above)
Comment (thibault-martinez) [reply bee-common/bee-packable/Cargo.toml]: Right

**Review by Adam-Gleave — COMMENTED** [review]
Looks good, I agree about `bee-packable-derive`, in case we write any other proc macros later on.

**Review by pvdrz — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**